### PR TITLE
Fix list view to not leak VT decorations

### LINKED ISSUE #17157: Formatting doesn't handle the property with ANSI escape sequence string value properly, which leads to color leaking
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PSReadLine)
- [X] Search the existing issues, especially the pinned issues.

### Exception report

```console
N/A
```

### Screenshot

Get-PSReadLineOption 1
Get-PSReadLineOption 2

### Environment data

```console
PS Version: 5.1.19041.1645
PS HostName: ConsoleHost
PSReadLine Version: 2.2.2
PSReadLine EditMode: Windows
OS: 10.0.19041.1 (WinBuild.160101.0800)
BufferWidth: 106
BufferHeight: 3000
```

### Steps to reproduce

Happens on PS 5.1 console and Windows Terminal app (cannot reproduce with the ISE nor PowerShell 7.2.2)
<br>

**To reproduce:**
1. Type `Get-PSReadlineOption | Format-List *`

<br>

**To resolve:**

1. Type `Get-PSReadLineOption` (without `| Format-List *`) and color scheme is restored after the property `CommandColor` is listed on-screen.  See bottom screenshot for a better explanation of this.

### Expected behavior

Color scheme should remain constant

### Actual behavior

- Console color scheme now appears like the `ListPredictionSelectedColor` color scheme
- Missing property values (see screenshot)
- Clearing the screen/changing console size doesn't resolve the issue.

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

Fix #17157

Fix list view to not leak VT decorations.
- Append 'reset' sequence when needed.
- Add test.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.

### CODE REVIEW

File: src/System.Management.Automation/FormatAndOutput/common/ListWriter.cs [hunk 1]
```diff
@@ -220,14 +228,30 @@ private void WriteSingleLineHelper(string prependString, string line, LineOutput
...
                 }
                 else
                 {
-                    lo.WriteLine(padding + PSStyle.Instance.Formatting.FormatAccent + PSStyle.Instance.Reset + sc[k]);
+                    _cachedBuilder
+                        .Append(padding)
+                        .Append(str);
                 }
+
+                if (str.Contains(ValueStringDecorated.ESC) && !str.EndsWith(PSStyle.Instance.Reset))
```
Comment (iSazonov) [inline src/System.Management.Automation/FormatAndOutput/common/ListWriter.cs:249]: What if the ESC value is escaped?

Comment (SeeminglyScience) [reply src/System.Management.Automation/FormatAndOutput/common/ListWriter.cs:249]: > What if the ESC value is escaped?

Can you escape a literal escape? Genuine question

Comment (daxian-dbw) [reply src/System.Management.Automation/FormatAndOutput/common/ListWriter.cs:249]: @iSazonov can you show an example of that? Currently, this check is at the core of VT handling in PowerShell (`$PSStyle` support depends on it).

Comment (iSazonov) [reply src/System.Management.Automation/FormatAndOutput/common/ListWriter.cs:249]: Backtick doesn't work at the level?

Comment (SeeminglyScience) [reply src/System.Management.Automation/FormatAndOutput/common/ListWriter.cs:249]: Nah it only works at the parser level to escape `` `e``. Once it's a literal `0x1b` char it doesn't do anything

**Review by SeeminglyScience — APPROVED** [review]
LGTM! ❤️

**Comment by adityapatwardhan** [comment]
Can we split the build.psm1 change into a different PR. We may want to backport this in the future. 

Also should the VT leak fix be taken to 7.2?

**Comment by daxian-dbw** [comment]
@adityapatwardhan Sure, will split the build.psm1 changes to a separate PR. I don't think this VT fix should be backported, this doesn't seem to be a big issue by telling from the report. But if we decided to take #17251, then this one should also be taken.

**Comment by daxian-dbw** [comment]
@adityapatwardhan The changes to `build.psm1` has been separated out.

**Pull request merged**